### PR TITLE
GDScript: Fix type inference for const reference to global class

### LINKED ISSUE #37529: Static type not recognized when class assigned to const (but works when assigned to var)
**Godot version:**
3.2.1 stable

**OS/device including version:**
Windows 10

**Issue description:**
A const can be assigned/initialized with a file class, but static type isn't inferred as I would expect. It does work as expected when a var is assigned/initialized with a class. See examples below.

**Steps to reproduce:**
```
# math.gd

class_name Math

static func sum(a: float, b: float) -> float:
	return a + b
```
Different files...
```
const math := Math

var c := math.sum(1.0, 2.0) # Causes error: "The assigned value does not have a set type..."
```
```
var math := Math

var c := math.sum(1.0, 2.0) # this is ok
```
```
const math := preload("res://math.gd")

var c := math.sum(1.0, 2.0) # this is ok
```
(I initially posted this in #33531, but I've come to believe it is a different issue.)

### PR DESCRIPTION
**Pull request by ThakeeNathees**
Fix: #37529
<s>(sorry for the commit message, I can't come up with one, any suggestions?)</s>

### CODE REVIEW

File: modules/gdscript/gdscript_parser.cpp [hunk 1]
```diff
@@ -837,6 +837,7 @@ GDScriptParser::Node *GDScriptParser::_parse_expression(Node *p_parent, bool p_s
 						if (scr.is_valid() && scr->is_valid()) {
 							ConstantNode *constant = alloc_node<ConstantNode>();
 							constant->value = scr;
+							constant->datatype = _type_from_variant(constant->value);
```
Comment (akien-mga) [inline modules/gdscript/gdscript_parser.cpp:840]: There are other code blocks nearby where a `ConstantNode` is allocated and the `datatype` member is not set. Should it always be set as here?

**Comment by akien-mga** [comment]
>  (sorry for the commit message, I can't come up with one, any suggestions?)

Well coming up with a descriptive commit message is quite important as it shows that you understand what changes you're making and why.

I'm not familiar with the parser to understand exactly what this code block handles, but going by the bug it's supposed to fix it could be something like:
```
GDScript: Fix type hints for const reference to global class

Fixes #37529.
```

File: modules/gdscript/gdscript_parser.cpp (hunk 1 above)
Comment (ThakeeNathees) [reply modules/gdscript/gdscript_parser.cpp:841]: most of the the constant's value(`Variant::Type`) is used for type checking, but It's essential to initialize the datatype as well.
@akien-mga

**Review by vnen — APPROVED** [review]
LGTM

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**